src/main/java/org/spongepowered/common/data/util/NbtDataUtil.java

```diff
@@ -196,6 +198,31 @@ public static void setLoreToNBT(ItemStack stack, List<Text> lore) {
         stack.getSubCompound(ITEM_DISPLAY, true).setTag(ITEM_LORE, list);
     }
 
+    public static boolean hasColorFromNBT(ItemStack stack) {
+        return stack.hasTagCompound() &&
+                stack.getTagCompound().hasKey(ITEM_DISPLAY) &&
+                stack.getTagCompound().getCompoundTag(ITEM_DISPLAY).hasKey(ITEM_COLOR);
+    }
+
+    public static Optional<Color> getColorFromNBT(NBTTagCompound subCompound) {
+        if (!subCompound.hasKey(ITEM_COLOR)) {
+            return Optional.empty();
+        }
+        return Optional.of(new Color(subCompound.getInteger(ITEM_COLOR)));
+    }
+
+    public static void removeColorFromNBT(ItemStack stack) {
+        if(stack.getSubCompound(ITEM_DISPLAY, false) == null) {
+            return;
+        }
+        stack.getSubCompound(ITEM_DISPLAY, false).removeTag(ITEM_COLOR);
+    }
+
+    public static void setColorToNbt(ItemStack stack, Color color) {
+        final int iColor = color.getRGB();
+        stack.getSubCompound(ITEM_DISPLAY, true).setInteger(ITEM_COLOR, iColor);
```

Comment:
And the other way round thry should use something like `DyeCole.from(Color)` as well.
If they have to use a valid DyeColor anyway to get the attached Color they could just take the DyeColor itself and set it.
IMO using Colors here is like using magic numbers. Some might work but most others will not. You need more context to know which ones will work. However that context alone is sufficient to get and set the values itself.
Just thing about the quota 0.00...01% will work. (I dont have a time for a calculator currently)